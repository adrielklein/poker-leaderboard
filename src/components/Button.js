import * as React from 'react';
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import SaveIcon from '@material-ui/icons/Save';
import CancelIcon from '@material-ui/icons/Cancel';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';

export const AddButton = ({ onExecute }) => (
  <div style={{ textAlign: 'center' }}>
    <Button
      color="primary"
      onClick={onExecute}
      title="Create new player"
    >
      New
    </Button>
  </div>
);

export const EditButton = ({ onExecute }) => (
  <IconButton onClick={onExecute} title="Edit player">
    <EditIcon />
  </IconButton>
);

export const DeleteButton = ({ onExecute }) => (
  <IconButton onClick={onExecute} title="Remove player">
    <DeleteIcon />
  </IconButton>
);

export const CommitButton = ({ onExecute }) => (
  <IconButton onClick={onExecute} title="Save changes">
    <SaveIcon />
  </IconButton>
);

export const CancelButton = ({ onExecute }) => (
  <IconButton color="secondary" onClick={onExecute} title="Cancel changes">
    <CancelIcon />
  </IconButton>
);
