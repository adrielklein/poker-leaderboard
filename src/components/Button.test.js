import React from 'react';
import { AddButton, EditButton, DeleteButton, CommitButton, CancelButton } from './Button';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

it('AddButton', () => {
  const component = shallow(<AddButton />);
  expect(toJson(component)).toMatchSnapshot();
});

it('EditButton', () => {
  const component = shallow(<EditButton />);
  expect(toJson(component)).toMatchSnapshot();
});

it('DeleteButton', () => {
  const component = shallow(<DeleteButton />);
  expect(toJson(component)).toMatchSnapshot();
});

it('CommitButton', () => {
  const component = shallow(<CommitButton />);
  expect(toJson(component)).toMatchSnapshot();
});

it('CancelButton', () => {
  const component = shallow(<CancelButton />);
  expect(toJson(component)).toMatchSnapshot();
});
