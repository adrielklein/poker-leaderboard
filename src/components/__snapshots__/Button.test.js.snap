// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <WithStyles(Button)
    color="primary"
    title="Create new player"
  >
    New
  </WithStyles(Button)>
</div>
`;

exports[`CancelButton 1`] = `
<WithStyles(IconButton)
  color="secondary"
  title="Cancel changes"
>
  <pure(CancelIcon) />
</WithStyles(IconButton)>
`;

exports[`CommitButton 1`] = `
<WithStyles(IconButton)
  title="Save changes"
>
  <pure(SaveIcon) />
</WithStyles(IconButton)>
`;

exports[`DeleteButton 1`] = `
<WithStyles(IconButton)
  title="Remove player"
>
  <pure(DeleteIcon) />
</WithStyles(IconButton)>
`;

exports[`EditButton 1`] = `
<WithStyles(IconButton)
  title="Edit player"
>
  <pure(EditIcon) />
</WithStyles(IconButton)>
`;
